### PR TITLE
Respond to AndroidView focus events.

### PR DESCRIPTION
**Pull request by amirh**
## Description

When an AndroidView gains focus we invoke the(newly introduced)
'TextInput.setPlatformViewClient' text_input system channel method
which sets the platform view as the text input target.

When the AndroidView loses focus we send a `clearFocus` message to
platform views system channel(so the engine can clear the focus from
the platform view).

This PR is going to land before the engine implementation is rolled to
the framework, we swallow MissingPluginException for the newly
introduced method channel methods so this is a no-op before the engine
is ready(after the engine is rolled with the corresponding change I'll
remove the logic to swallow the exceptions).

The engine counterpart is in: https://github.com/flutter/engine/pull/9203

## Related Issues

#19718

## Tests

I added the following tests:

* AndroidView sets a platform view text input client when focused
* AndroidView clears platform focus when unfocused

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*
- [x] No, this is *not* a breaking change.

### CODE REVIEW

**Comment by amirh** [comment]
cc @jason-simmons (for context when reviewing the corresponding engine PR)

**Comment by phanirithvij** [comment]
Please merge this as soon as possible.
I've been waiting for this to get fixed as we're planning on using Webviews for login and signup in our application.
And this is what's stopping us from using flutter.
I don't think we can wait until October.

File: packages/flutter/test/services/fake_platform_views.dart [hunk 1]
```diff
@@ -154,6 +158,19 @@ class FakeAndroidPlatformViewsController {
...
     return Future<dynamic>.sync(() => null);
   }
+
+  Future<dynamic> _clearFocus(MethodCall call) {
+    final int id = call.arguments;
+
+    if (!_views.containsKey(id))
+      throw PlatformException(
+        code: 'error',
+        message: 'Trying to dispose a platform view with unknown id: $id',
```
Comment (gspencergoog) [inline packages/flutter/test/services/fake_platform_views.dart]:
```suggestion
        message: 'Trying to clear the focus on a platform view with unknown id: $id',
```

**Review by gspencergoog — APPROVED** [review]
32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d

**Comment by amirh** [comment]
Waiting for flutter/engine#9203 to be approved before landing this one.
(This PR will land before the engine PR, but I'm waiting to see the method channel protocol is approved on both sides)

**Comment by sakina1403** [comment]
> Waiting for [flutter/engine#9203](https://github.com/flutter/engine/pull/9203) to be approved before landing this one.
> (This PR will land before the engine PR, but I'm waiting to see the method channel protocol is approved on both sides)

Is there an ETA??? Waiting desperately for this since March. Tired of using workarounds. We really love Flutter but this has been holding me back since a long time. Our App has heavy use of WebView and a lot of JS communication as well and this is the only plugin perfect for postMessage Communication. We have 100K+ downloads but usability is affecting largely because of this issue.

Thank you for all your efforts  @amirh . Unless and until it's reaally necessary to wait please please merge this ASAP.

**Pull request merged**